packages/core/bootstrap/src/lib/server.ts

```diff
@@ -120,23 +111,25 @@ export const initHandler =
       process.exit()
     })
 
+    app.addHook('onClose', async () => {
+      storeSlice('cacheWarmer').dispatch(warmupShutdown())
+      storeSlice('errorBackoff').dispatch(shutdown())
+      storeSlice('ws').dispatch(WSReset())
+      context.cache?.instance?.close()
+    })
+
     return new Promise((resolve) => {
-      const server = app.listen(port, () => {
-        server.on('close', () => {
-          storeSlice('cacheWarmer').dispatch(warmupShutdown())
-          storeSlice('errorBackoff').dispatch(shutdown())
-          storeSlice('ws').dispatch(WSReset())
-          context.cache?.instance?.close()
-        })
-
-        logger.info(`Listening on port ${(server.address() as AddressInfo).port}!`)
-        resolve(server)
+      app.listen(port, (_, address) => {
+        logger.info(`Listening on port ${address}!`)
```

Comment:
Haven't actually checked, but this (`on port ${address}`) seems incorrect? Should it be using the `port` var instead?

Reply:
yes! its wrong :) done!